model/simulationtests/python_plugin_search_paths.py

```diff
@@ -0,0 +1,132 @@
+        super().__init__()
+        self.do_setup = True
+
+    def on_end_of_zone_timestep_before_zone_reporting(self, state) -> int:
+        if self.do_setup:
+            self.data['zone_volumes'] = []
+            self.data['zone_temps'] = []
+            zone_names = {zone_names_str_list}
+            for zone_name in zone_names:
+                handle = self.api.exchange.get_internal_variable_handle(state, 'Zone Air Volume', zone_name)
+                zone_volume = self.api.exchange.get_internal_variable_value(state, handle)
+                self.data['zone_volumes'].append(zone_volume)
+                self.data['zone_temps'].append(
+                    self.api.exchange.get_variable_handle(state, 'Zone Mean Air Temperature', zone_name)
+                )
+            self.data['avg_temp_variable'] = self.api.exchange.get_global_handle(state, '{py_var.nameString()}')
+            self.data['trend'] = self.api.exchange.get_trend_handle(state, '{py_trend_var.nameString()}')
+            self.data['running_avg_temp_variable'] = self.api.exchange.get_global_handle(state, '{py_var2.nameString()}')
+            self.do_setup = False
+        zone_temps = list()
+        for t_handle in self.data['zone_temps']:
+            zone_temps.append(self.api.exchange.get_variable_value(state, t_handle))
+        numerator = 0.0
+        denominator = 0.0
+        for i in range(len(self.data['zone_volumes'])):
+            numerator += self.data['zone_volumes'][i] * zone_temps[i]
+            denominator += self.data['zone_volumes'][i]
+        average_temp = numerator / denominator
+        self.api.exchange.set_global_value(state, self.data['avg_temp_variable'], average_temp)
+
+        past_daily_avg_temp = self.api.exchange.get_trend_average(state, self.data['trend'], {n_timesteps})
+        self.api.exchange.set_global_value(state, self.data['running_avg_temp_variable'], past_daily_avg_temp)
+        return 0
+"""
+
+# Write it to a temporary directory so we don't pollute the current directory
+# ExternalFile will copy it
+pluginPath = Path(tempfile.gettempdir()) / f"{Path(__file__).stem}_program.py"
+pluginPath.write_text(python_plugin_file_content)
+
+# create the external file object
+external_file = openstudio.model.ExternalFile.getExternalFile(model, str(pluginPath))
+external_file = external_file.get()
+
+# create the python plugin instance object
+python_plugin_instance = openstudio.model.PythonPluginInstance(external_file, pluginClassName)
+python_plugin_instance.setRunDuringWarmupDays(False)
+
+# create the python plugin search paths object (this test should fail without it)
+python_plugin_search_paths = model.getPythonPluginSearchPaths()
+python_plugin_search_paths.setAddCurrentWorkingDirectorytoSearchPath(True)
+python_plugin_search_paths.setAddInputFileDirectorytoSearchPath(True)
+python_plugin_search_paths.setAddepinEnvironmentVariabletoSearchPath(True)
+python_plugin_search_paths.addSearchPath(str(Path(__file__).parent))
```

Comment:
I added a python version. Really the fact that this takes only a string bothers me.